File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;  // Return 0 for empty input
        
        sort(nums.begin(), nums.end());
        
        int current = 1;
        int longest = 1;
        
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1]) {
                // Skip duplicates
                continue;
            } else if (nums[i] == nums[i - 1] + 1) {
                // Consecutive element
                current++;
            } else {
                // Not consecutive, update longest and reset current
                longest = max(longest, current);
                current = 1;
            }
        }
        
        // Final check to update longest
        longest = max(longest, current);
        
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
    Arrays.sort(nums);
    if(nums.length==0){
        return 0;
    }
       int current=1;
       int longest=1;
       //checking for duplicates
       for(int i=1; i<nums.length; i++){
        if(nums[i]==nums[i-1]){
            continue;
        }
        else if(nums[i]==nums[i-1]+1){
            current++;
        }
        else{
            longest=Math.max(longest, current);
            current=1;
        }

       }
        longest=Math.max(longest, current); 
       return longest;
    }
}